src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -126,7 +125,12 @@ export default function Swap({
   } = useDerivedSwapInfo()
 
   // detects trade load
-  const quoteLoading = useIsQuoteLoading()
+  const [quote, quoteLoading] = useGetQuoteAndStatus({ token: INPUT.currencyId, chainId })
+
+  // major quote load indication (param change, token change etc)
+  // opposite could be just a simple timeout check
+  const isQuoteHardLoading = quoteLoading && !quote?.price.amount
+  const isQuoteSoftLoading = quoteLoading && !isQuoteHardLoading
```

Comment:
Is this something we should return already from the hook? 

If we want to use this elsewhere we wouldn't need to replicate the logic.